localstack/services/ec2/provider.py

```diff
@@ -97,25 +97,27 @@ def describe_availability_zones(
         describe_availability_zones_request: DescribeAvailabilityZonesRequest,
     ) -> DescribeAvailabilityZonesResult:
         backend = get_ec2_backend(context.account_id, context.region)
-
-        availability_zones = []
         zone_names = describe_availability_zones_request.get("ZoneNames")
-        if zone_names:
-            for zone in zone_names:
-                zone_detail = backend.get_zone_by_name(zone)
-                if zone_detail:
+        zone_ids = describe_availability_zones_request.get("ZoneIds")
+        if zone_names or zone_ids:
+            filters = {
+                "zone-name": zone_names if zone_names else None,
+                "zone-id": zone_ids if zone_ids else None,
```

Comment:
These are redundant

```suggestion
                "zone-name": zone_names,
                "zone-id": zone_ids,
```

Reply:
Hey @viren-nadkarni ,

Thank for the suggestion. I have made the required change.